plugins/scanners/fossid/src/main/kotlin/FossId.kt

```diff
@@ -236,21 +236,24 @@ class FossId internal constructor(
     private fun createSingleIssueSummary(startTime: Instant, issue: Issue) =
         ScanSummary.EMPTY.copy(startTime = startTime, endTime = Instant.now(), issues = listOf(issue))
 
-    override fun scanPackage(pkg: Package, nestedProvenance: NestedProvenance?, context: ScanContext): ScanResult {
+    override fun scanPackage(nestedProvenance: NestedProvenance?, context: ScanContext): ScanResult {
         val startTime = Instant.now()
 
-        // FossId actually never uses the provenance determined by the scanner, but determines the source code to
-        // download itself based on the passed VCS information.
-        val provenance = pkg.vcsProcessed.revision.takeUnless { it.isBlank() }
-            ?.let { RepositoryProvenance(pkg.vcsProcessed, it) } ?: UnknownProvenance
```

Comment:
I don't like removing the `pkg` argument here for two reasons:
* Previously, FossId always used the VCS information, now it only does so if the resolved provenance is a `RepositoryProvenance`. That means that if someone is using FossId to scan packages this does not work properly anymore if they configure the source code origin priority to prioritize source artifacts over VCS.
* There could be package scanners that use other input than the provenance, like the package id or Purl. For example, doesn't clearly defined scan based on identifiers instead of provenance?

Reply:
> Previously, FossId always used the VCS information

Good point, FossId never adhered to the configured `sourceCodeOrigin`.

> There could be package scanners that use other input than the provenance, like the package id or Purl.

That's actually exactly what the Double Open scanner does. But still the reference `pkg` is not needed, as `context.coveredPackages` is enought (and also better, in the sense that the packages still contains their VCS path).

So, I could solve both issues by operating on `context.coveredPackages`, and still remove the "artificial" `pkg` parameter. I really don't like it as it does reflect a real package when it comes from a VCS with a path.

Reply:
That sounds good to me.